### ISSUE TITLE
Masterkeys Pro M Support

### ISSUE THREAD

**Issue by MalcolmLorber** [antonpup/Aurora#603]
Are there plans to add support for the CoolerMaster Masterkeys Pro M RGB model? I see that the L and S are already supported, and they use the same SDK.

**Comment by brainbug89** [comment]
What happens when you're using the L layout?

**Comment by MalcolmLorber** [comment]
I believe the Cooler Master SDK wont do anything if the wrong device is selected (from some quick testing just now). I don't know if something is supposed to be happening, but nothing does. They keyboard retains its original lighting.

**Comment by brainbug89** [comment]
You're right. `CoolerMasterSDK.SetControlDevice(CoolerMasterSDK.DEVICE_INDEX.DEV_MKeys_M);` needs to be added to https://github.com/antonpup/Aurora/blob/dev/Project-Aurora/Project-Aurora/Devices/CoolerMaster/CoolerMasterDevice.cs

**Comment by brainbug89** [comment]
I have added Masterkeys Pro M and Masterkeys Pro S White to the CoolerMasterSDK but I don't own a CM device. So someone else needs to test it. @MalcolmLorber could you try this [built](https://www.dropbox.com/s/4otrtrj1y0rlpz4/Aurora_MasterkeysProM.zip?dl=0)?

**Comment by MalcolmLorber** [comment]
The build you linked doesn't have them as option in the device list.

**Comment by brainbug89** [comment]
I know. I didn't added the layout yet. Just try the L layout. The CoolerMasterSDK should get initialized.

**Comment by MalcolmLorber** [comment]
Well I'm not sure what should happen, but nothing does happen. What should I be looking for?

**Comment by brainbug89** [comment]
My fault, I forgot to replace the SDK.dll I'll do this tomorow and send you a new built.

**Comment by MalcolmLorber** [comment]
Replacing the dll with the latest causes lighting to work (except for layout issues), though the program does crash the moment it loses focus.

**Comment by brainbug89** [comment]
Can you provide a log?

**Comment by MalcolmLorber** [comment]
Where do I find the log?

**Comment by brainbug89** [comment]
You can find the log files here: ` %appdata%/Aurora/logs/`

**Comment by MalcolmLorber** [comment]
Attached it
[2017_15_06.txt](https://github.com/antonpup/Aurora/files/1079365/2017_15_06.txt)

**Comment by brainbug89** [comment]
```
[ERROR] 6/15/2017 8:00:52 PM: Exception in GetActiveWindowsProcessnameSystem.ComponentModel.Win32Exception (0x80004005): A 32 bit processes cannot access modules of a 64 bit process.
   at System.Diagnostics.NtProcessManager.GetModuleInfos(Int32 processId, Boolean firstModuleOnly)
   at System.Diagnostics.NtProcessManager.GetFirstModuleInfo(Int32 processId)
   at System.Diagnostics.Process.get_MainModule()
   at Aurora.Utils.ActiveProcessMonitor.GetActiveWindowsProcessname() in D:\git\Aurora\Project-Aurora\Project-Aurora\Utils\ActiveProcessMonitor.cs:line 88
```
Seems to be a problem with something else. :thinking:

**Comment by simon-wh** [comment]
Run Aurora as admin.

**Comment by MalcolmLorber** [comment]
Running as admin doesn't change anything, same error.

**Comment by simon-wh** [comment]
I didn't process the error properly. The build you're using was built as
32bit or AnyCPU when it needs to be built as x64.

**Comment by simon-wh** [comment]
Fixed due to PR #608. Thanks @brainbug89!
